packages/router/src/adapters/contract/contract.ts

```diff
@@ -44,6 +45,197 @@ export const removeLiquidityEvt = new Evt<{
 export const isChainSupportedByGelato = _isChainSupportedByGelato;
 export const gelatoSend = _gelatoSend;
 
+/**
+ * Helper method for waiting for relayer transaction go through using event handling. Retrieves the receipt afterwards
+ * and increments the amount spent on relayer fees.
+ *
+ * @param chainId - Chain ID of the transaction.
+ * @param transactionId - Transaction ID of the transaction to wait for.
+ * @param evt - Event to wait for.
+ * @param relayerFee - Relayer fee amount.
+ * @param relayerFeeAsset - Relayer fee asset ID.
+ * @param reason - Reason for the transaction (i.e. transaction type).
+ * @param requestContext - Request context.
+ * @returns TransactionReceipt of the transaction (if it's successful).
+ * @throws Timeout error if the event is not emitted within the timeout period.
+ */
+const waitForRelayer = async (
+  chainId: number,
+  transactionId: string | undefined,
+  evt: Evt<any>,
+  relayerFee: {
+    asset: string;
+    amount: string;
+  },
+  reason: TransactionReason,
+  requestContext: RequestContext,
+) => {
+  const { txService } = getContext();
+  // Listen for event on contract.
+  const { event } = transactionId
+    ? await evt.pipe(({ args }) => args.txData.transactionId === transactionId).waitFor(300_000)
+    : await evt.waitFor(300_000);
```

Comment:
Waiting for just any event wouldn't work here, you still need something to match as an identifier -- none of our evts are that specific

Reply:
Ahh I see, you are handling the `removeLiquidity` in the `:` block. Since that is the only reason where a call wouldn't have a `transactionId`, can we handle the `undefined` case more explicitly?